src/ProductConstructionService/ProductConstructionService.FeedCleaner/FeedCleanerJob.cs

```diff
@@ -45,61 +44,104 @@ public async Task CleanManagedFeedsAsync()
         {
             _logger.LogInformation("Processing feeds for {account}...", azdoAccount);
 
-            List<AzureDevOpsFeed> allFeeds;
-            try
+            var (packageFeeds, symbolFeeds) = await FetchFeeds(azdoAccount);
+
+            _logger.LogInformation("Processing {feedCount} package feeds...",
+                packageFeeds.Count);
+
+            int feedsCleaned = await ProcessFeedsInParallelAsync(packageFeeds, async (scope, feed) =>
             {
-                allFeeds = await _azureDevOpsClient.GetFeedsAsync(azdoAccount);
-            }
-            catch (Exception ex)
+                var feedCleaner = scope.ServiceProvider.GetRequiredService<FeedCleaner>();
+                await feedCleaner.CleanFeedAsync(feed);
+                await Task.CompletedTask;
+            });
+
+            _logger.Log(
+                feedsCleaned != packageFeeds.Count ? LogLevel.Warning : LogLevel.Information,
+                "Successfully processed {count}/{totalCount} package feeds for {account}",
+                feedsCleaned,
+                packageFeeds.Count,
+                azdoAccount);
+
+            _logger.LogInformation("Processing {feedCount} symbol feeds...",
+                symbolFeeds.Count);
+
+            // We refresh the feed information so that symbol feeds can be cleaned based on up-to-date package feeds
+            (packageFeeds, symbolFeeds) = await FetchFeeds(azdoAccount);
+
+            feedsCleaned = await ProcessFeedsInParallelAsync(packageFeeds, async (scope, feed) =>
```

Comment:
I'm not sure what, exactly, but something is wrong here since we're not using the `symbolFeeds` anywhere

Reply:
We only clean the symbol feeds below. I used the `_` to make it obvious

Reply:
but shouldn't this be 
```suggestion
            feedsCleaned = await ProcessFeedsInParallelAsync(symbolFeeds, async (scope, feed) =>
```
so we go over the symbol feeds, and delete what we need to?

Reply:
Oh, I see what you mean, good catch